internal/incident/incident.go

```diff
@@ -300,172 +338,116 @@ func (i *Incident) evaluateEscalations() {
+						return err
 					}
 				}
 			}
 		}
 	}
 
-	for recipientKey, state := range i.Recipients {
-		r := i.runtimeConfig.GetRecipient(recipientKey)
-		if r == nil {
-			continue
-		}
+	return nil
+}
 
-		isEscalationRecipient := escalationRecipients[recipientKey]
-		if !isEscalationRecipient && (!managed || state.Role > RoleRecipient) {
-			for _, contact := range r.GetContactsAt(ev.Time) {
-				if contactChannels[contact] == nil {
-					contactChannels[contact] = make(map[int64]struct{})
-				}
-				contactChannels[contact][contact.DefaultChannelID] = struct{}{}
-			}
+// notifyContacts executes all the given pending notifications of the current incident.
+// Returns error on database failure or if the provided context is cancelled.
+func (i *Incident) notifyContacts(ctx context.Context, ev *event.Event, notifications []*NotificationEntry) error {
+	for _, notification := range notifications {
+		contact := i.runtimeConfig.Contacts[notification.ContactID]
+
+		if i.notifyContact(contact, ev, notification.ChannelID) != nil {
+			notification.State = NotificationStateFailed
+		} else {
+			notification.State = NotificationStateSent
 		}
-	}
 
-	for contact, channels := range contactChannels {
-		hr := &HistoryRow{
-			Key:                       recipient.ToKey(contact),
-			EventID:                   utils.ToDBInt(ev.ID),
-			Time:                      types.UnixMilli(time.Now()),
-			Type:                      Notified,
-			CausedByIncidentHistoryID: causedBy,
+		notification.SentAt = types.UnixMilli(time.Now())
+		stmt, _ := i.db.BuildUpdateStmt(notification)
+		if _, err := i.db.NamedExecContext(ctx, stmt, notification); err != nil {
+			i.logger.Errorw(
+				"Failed to update contact notified incident history", zap.String("contact", contact.String()),
+				zap.Error(err),
+			)
 		}
 
 		if err := ctx.Err(); err != nil {
 			return err
 		}
+	}
 
-		for chId := range channels {
-			ch := i.runtimeConfig.Channels[chId]
-			if ch == nil {
-				i.logger.Errorw("Could not find config for channel", zap.Int64("channel_id", chId))
-				continue
-			}
+	return nil
+}
 
-			chType := ch.Type
+// notifyContact notifies the given recipient via a channel matching the given ID.
+func (i *Incident) notifyContact(contact *recipient.Contact, ev *event.Event, chID int64) error {
+	ch := i.runtimeConfig.Channels[chID]
+	if ch == nil {
+		i.logger.Errorw("Could not find config for channel", zap.Int64("channel_id", chID))
 
-			i.logger.Infof("Notify contact %q via %q of type %q", contact.FullName, ch.Name, chType)
+		return errors.New(fmt.Sprintf("could not find config for channel ID: %d", chID))
+	}
 
-			hr.ChannelID = utils.ToDBInt(ch.ID)
+	chType := ch.Type
 
-			_, err := i.AddHistory(ctx, tx, hr, false)
-			if err != nil {
-				i.logger.Errorw(
-					"Failed to insert contact notified incident history", zap.String("contact", contact.String()),
-					zap.Error(err),
-				)
-			}
+	i.logger.Infof("Notify contact %q via %q", contact.FullName, chType)
```

Comment:
> If there was a need to shorten the message, it should probably state the name instead of the type as there can be multiple configured channels of the same type, so the name provides more information.

This is due to the previous version of the current implementation, as there was no way to access the full channel configuration, but then forgot to adjust it accordingly.